### PR TITLE
Sid Mann

### PR DESCRIPTION
**Pull request by sjmann2**
I struggled with knowing the best place to write tests for edge cases. I wrote some of them in the features spec when that felt relevant and others in the facade spec. The other thing I struggled with was the search_by_keyword method in the facade, and how to create a quote object that had both the results data and the total count in it. I ended up putting these in two separate methods but I want to go back and refactor to get the total count instantiated in a quote object.

### CODE REVIEW

File: app/facades/quotes_facade.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+    quotes_data = QuotesService.search_by_keyword(keyword)
+    quotes = quotes_data[:results].map do |data|
+      Quote.new(data)
+    end
+  end
+
+  def self.total_count_by_keyword(keyword)
+    quotes_data = QuotesService.search_by_keyword(keyword)
+    quotes_data[:totalCount]
+  end
```
Comment (epintozzi) [inline app/facades/quotes_facade.rb:12]: You noted this in your submission comments, but you are hitting the API endpoint twice and getting the same result both times since the calls are identical. This is not very efficient. Something you could do to return both the count and the collection of quotes from the same method would be to return an array that contains each of those at the end of the `search_by_keyword` method. And then assign to variables in the controller. Something like this:
```ruby
  def self.search_by_keyword(keyword)
    quote_data = QuotesService.search_by_keyword(keyword)
    quotes = quote_data[:results].map do |data|
      Quote.new(data)
    end
    [quote_data[:totalCount], quotes]
  end
```

And then you could update the controller to:
```ruby
def index
    @total_count, @quotes = QuotesFacade.search_by_keyword(params[:query])
    @keyword = params[:query]
  end
```

File: spec/poros/quote_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe Quote do
+  before :each do 
+    @quote = QuotesFacade.search_by_keyword("I am").first
```
Comment (epintozzi) [inline spec/poros/quote_spec.rb:5]: This definitely works, but there are a couple things I don't love about the approach are using the facade to get the data to send to the poro. 1) If your facade ever changes (ex: the name of the method), then this test breaks even though it's unrelated. 2) It's making an API call when all we really need to test this is some sample data in the format that the `Quote` class is expecting. We don't need to use VCR or other mocking tools to do this. 3) If the first result from the API ever changes, this test fails. We can just make up data ourselves and do something like this instead:
```ruby
  before :each do
    @quote = Quote.new({ content: "If I am not for myself, who will be for me? If I am not for others, what am I? And if not now, when?",
                         author: "Rabbi Hillel",
                         tags: ["famous-quotes", "wisdom"] })
  end
```

**Review by epintozzi — COMMENTED** [review]
Overall great work. You did a nice job testing your happy and sad paths. The code is easy to read and follow. I left a suggestion for how you can address the issue in the facade that you noted in your submission as well as one thought on a different setup for your poro spec. Nice work!
